packages/runtime/container-runtime/src/opLifecycle/batchManager.ts

```diff
@@ -169,14 +174,20 @@ export class BatchManager {
 		const startPoint = this.pendingBatch.length;
 		return {
 			rollback: (process: (message: BatchMessage) => void) => {
-				for (let i = this.pendingBatch.length; i > startPoint; ) {
-					i--;
-					const message = this.pendingBatch[i];
+				const rollbackOps = this.pendingBatch.splice(startPoint).reverse();
+				for (const message of rollbackOps) {
```

Comment:
Consider renaming `rollbackOps` to `lifoRollbackOps` or `rollbackOpsLifo` or add a comment to make it super clear (the `reverse` could be missed at a glance)